### ISSUE TITLE
Add support for content all content types

### ISSUE THREAD

**Issue by ckeboss** [cdimascio/express-openapi-validator#143]
I noticed the `TODO` comment for `add support for content all content types`.

Is there a current idea on how to do this? I would be happy to do the work.

**Comment by cdimascio** [comment]
awesome! i very much welcome the help. thanks

all in all, this area hasn't been deeply explored. some types may just work, others may require a bunch of work.

as a starting point, have a look https://swagger.io/specification/#mediaTypeObject, particularly the Media Type Object and Media Type Examples sections

as for ways to get started:
- create some tests using different mime types and observe how the validator responds
- review some places in the code that lightly test media types other than json e.g. text/plain [here](https://github.com/cdimascio/express-openapi-validator/blob/bbcd5aed1f8e3fd872b1c88ac6756c572c4d7750/test/request.bodies.ref.spec.ts#L44). 
- multipart/formdata has some [tests](https://github.com/cdimascio/express-openapi-validator/blob/bbcd5aed1f8e3fd872b1c88ac6756c572c4d7750/test/multipart.spec.ts) due to file upload support
- binary media types - file upload support may need some enhancements as we currently hardcode `multipart/formdata` [here](https://github.com/cdimascio/express-openapi-validator/blob/bbcd5aed1f8e3fd872b1c88ac6756c572c4d7750/src/middlewares/openapi.multipart.ts#L54). For example, some modifications may be needed to support the following:
```
requestBody:
  content:
      # a binary file of type png or jpeg
    'image/jpeg':
      schema:
        type: string
        format: binary
    'image/png':
      schema:
        type: string
        format: binary     
```
-  application/x-www-form-urlencoded. does this just work? does it need work? note we hard code `application/json` in a few places ([1](https://github.com/cdimascio/express-openapi-validator/blob/bbcd5aed1f8e3fd872b1c88ac6756c572c4d7750/src/middlewares/ajv/index.ts#L103), [2](https://github.com/cdimascio/express-openapi-validator/blob/bbcd5aed1f8e3fd872b1c88ac6756c572c4d7750/src/middlewares/openapi.request.validator.ts#L331), [3](https://github.com/cdimascio/express-openapi-validator/blob/bbcd5aed1f8e3fd872b1c88ac6756c572c4d7750/src/middlewares/openapi.response.validator.ts#L105)) which may need to change
```
    application/x-www-form-urlencoded:
      schema:
        type: object
        properties:
          id:
            type: string
            format: uuid
          address:
            # complex types are stringified to support RFC 1866
            type: object
            properties: {}
```

**Comment by ckeboss** [comment]
Looking further into this, I may have bitten off a bit more than I can chew for supporting all media types.

What I did do, as it's what my use-case needed, is add support for any `json` media type, allowing for parsing anything with the `+json` suffex, as defined in https://tools.ietf.org/html/rfc6838.

**Comment by cdimascio** [comment]
this is available in v3.2.0. much thanks to @ckeboss!
